crates/rethnet_eth/src/spec.rs

```diff
@@ -0,0 +1,180 @@
+
+    CONFIG.get_or_init(|| {
+        let hardforks = ROPSTEN_HARDFORKS
+            .iter()
+            .map(|(block_number, spec_id)| (U256::from(*block_number), *spec_id))
+            .collect();
+
+        ChainConfig {
+            name: "ropsten".to_string(),
+            hardforks,
+        }
+    })
+}
+
+const RINKEBY_HARDFORKS: &[(u64, SpecId)] = &[
+    (1_035_301, SpecId::BYZANTIUM),
+    (3_660_663, SpecId::CONSTANTINOPLE),
+    (4_321_234, SpecId::PETERSBURG),
+    (5_435_345, SpecId::ISTANBUL),
+    (8_290_928, SpecId::BERLIN),
+    (8_897_988, SpecId::LONDON),
+];
+
+fn rinkeby_config() -> &'static ChainConfig {
+    static CONFIG: OnceLock<ChainConfig> = OnceLock::new();
+
+    CONFIG.get_or_init(|| {
+        let hardforks = RINKEBY_HARDFORKS
+            .iter()
+            .map(|(block_number, spec_id)| (U256::from(*block_number), *spec_id))
+            .collect();
+
+        ChainConfig {
+            name: "rinkeby".to_string(),
+            hardforks,
+        }
+    })
+}
+
+const GOERLI_HARDFORKS: &[(u64, SpecId)] = &[
+    (1_561_651, SpecId::ISTANBUL),
+    (4_460_644, SpecId::BERLIN),
+    (5_062_605, SpecId::LONDON),
+];
+
+fn goerli_config() -> &'static ChainConfig {
+    static CONFIG: OnceLock<ChainConfig> = OnceLock::new();
+
+    CONFIG.get_or_init(|| {
+        let hardforks = GOERLI_HARDFORKS
+            .iter()
+            .map(|(block_number, spec_id)| (U256::from(*block_number), *spec_id))
+            .collect();
+
+        ChainConfig {
+            name: "goerli".to_string(),
+            hardforks,
+        }
+    })
+}
+
+const KOVAN_HARDFORKS: &[(u64, SpecId)] = &[
+    (5_067_000, SpecId::BYZANTIUM),
+    (9_200_000, SpecId::CONSTANTINOPLE),
+    (10_255_201, SpecId::PETERSBURG),
+    (14_111_141, SpecId::ISTANBUL),
+    (24_770_900, SpecId::BERLIN),
+    (26_741_100, SpecId::LONDON),
+];
+
+fn kovan_config() -> &'static ChainConfig {
+    static CONFIG: OnceLock<ChainConfig> = OnceLock::new();
+
+    CONFIG.get_or_init(|| {
+        let hardforks = KOVAN_HARDFORKS
+            .iter()
+            .map(|(block_number, spec_id)| (U256::from(*block_number), *spec_id))
+            .collect();
+
+        ChainConfig {
+            name: "kovan".to_string(),
+            hardforks,
+        }
+    })
+}
+
+fn chain_configs() -> &'static HashMap<U256, &'static ChainConfig> {
+    static CONFIGS: OnceLock<HashMap<U256, &'static ChainConfig>> = OnceLock::new();
+
+    CONFIGS.get_or_init(|| {
+        let mut hardforks = HashMap::new();
+        hardforks.insert(U256::from(1), mainnet_config());
+        hardforks.insert(U256::from(3), ropsten_config());
+        hardforks.insert(U256::from(4), rinkeby_config());
+        hardforks.insert(U256::from(5), goerli_config());
+        hardforks.insert(U256::from(42), kovan_config());
+
+        hardforks
+    })
+}
+
+/// Returns the `SpecId` corresponding to the provided chain ID and block number, if the chain ID is supported.
```

Comment:
With just backticks, `cargo doc` won't link this to the doc for that datatype. I tried it (before and after my suggested change), and I confirmed it in https://doc.rust-lang.org/rustdoc/write-documentation/linking-to-items-by-name.html#linking-to-items-by-name

```suggestion
/// Returns the [SpecId] corresponding to the provided chain ID and block number, if the chain ID is supported.
```

Reply:
Good catch! Backticks still add the formatting though, right?

Reply:
yup just not the link